src/lib/swapper/swappers/ThorchainSwapper/utils/makeSwapMemo/assertIsValidMemo.ts

```diff
@@ -70,6 +74,34 @@ export const assertIsValidMemo = (memo: string): void => {
     })
   }
 
+  // Check if limit is a valid number
+  if (isNaN(parseFloat(limit)) || !isFinite(parseFloat(limit))) {
```

Comment:
Use `bigbumber.js` instead? isNaN behavior is scary, it doesn't return whether a number is not a number, but rather whether or not `Number(number) === NaN`

<img width="101" alt="image" src="https://user-images.githubusercontent.com/17035424/235145381-d5c3920b-bc47-495e-ab28-7748c9176bea.png">

Reply:
(or `bignumber.js` if you feel like it)

Reply:
For sure - used `bn` in https://github.com/shapeshift/web/pull/4353/commits/ed83875e0eabedd9af6e6b58476170cb0e63bb5e.